### PR TITLE
Make settlements client aware and add chargeback pagination

### PR DESCRIPTION
**Pull request by Smitsel**

### CODE REVIEW

**Comment by willemstuursma** [comment]
This one is also used for the top level resource right?

**Comment by Smitsel** [comment]
Yes for toplevel, and also $payment->chargebacks() and $settlement->chargebacks()

**Comment by willemstuursma** [comment]
It looks fine but I have a feeling of code duplication. Can you clean that up by introducing a trait or an abstract class?

**Comment by Smitsel** [comment]
Cleaned it up by expanding the abstract CursorCollection.
Great suggestion!

File: src/Resources/CursorCollection.php [hunk 1]
```diff
@@ -14,26 +14,63 @@
     /**
      * @param MollieApiClient $client
      * @param int $count
-     * @param object $_links
+     * @param object[] $_links
      */
-    public function __construct(MollieApiClient $client, $count, $_links)
+    final public function __construct(MollieApiClient $client, $count, $_links)
```
Comment (willemstuursma) [inline src/Resources/CursorCollection.php]: Make it protected, it's only instanciated by this class right?

File: src/Resources/CursorCollection.php [hunk 2]
```diff
@@ -14,26 +14,63 @@
...
 
         $this->client = $client;
     }
 
+    /**
+     * Return the resource object
+     *
+     * @return BaseResource
+     */
+    abstract protected function getResourceObject();
```
Comment (willemstuursma) [inline src/Resources/CursorCollection.php]: Maybe better call this `createResourceObject` because it has to create new ones.

**Comment by willemstuursma** [comment]
This looks great.

File: src/Resources/CursorCollection.php (hunk 1 above)
Comment (Smitsel) [reply src/Resources/CursorCollection.php]: It extends from BaseCollection (the one without pagination) and ArrayObject.

**Pull request merged**